src/julia_gc.c

```diff
@@ -773,6 +777,19 @@ void GAP_InitJuliaMemoryInterface(jl_module_t *   module,
     jl_gc_set_cb_post_gc(PostGCHook, 1);
     // jl_gc_enable(0); /// DEBUGGING
 
+#if defined(USE_GAP_INSIDE_JULIA)
+    DatatypeGapObj =
+        (jl_datatype_t *)jl_get_global(module, jl_symbol("GapObj"));
+    jl_reinit_foreign_type(DatatypeGapObj, MPtrMarkFunc, NULL);
+
+    DatatypeSmallBag =
+        (jl_datatype_t *)jl_get_global(module, jl_symbol("SmallBag"));
+    jl_reinit_foreign_type(DatatypeSmallBag, BagMarkFunc, JFinalizer);
+
+    DatatypeLargeBag =
+        (jl_datatype_t *)jl_get_global(module, jl_symbol("LargeBag"));
+    jl_reinit_foreign_type(DatatypeLargeBag, BagMarkFunc, JFinalizer);
+#else
     if (module == 0) {
```

Comment:
Inside GAP we know that both `module` and `parent` are unused, so these checks could be removed.

Alternatively, you could change `GAP_InitJuliaMemoryInterface(0, 0);` to `GAP_InitJuliaMemoryInterface(jl_main_module, jl_any_type);`